### ISSUE TITLE
Search by folder name should be case insensitive

### ISSUE THREAD

**Issue by Sylvain-L** [Microsoft/AppCenter-SDK-React-Native#212]
Maybe it's bad practice on our part, but I thought I should share this here.

When using react-native link the package.json is used to guess the folder name containing the app code.

```javascript
const pjson = require(path.join(process.cwd(), './package.json'));

const appDelegatePaths = glob.sync('**/AppDelegate.m', { ignore: 'node_modules/**' });
const appDelegatePath = findFileByAppName(appDelegatePaths, pjson ? pjson.name : null) || appDelegatePaths[0];
```

`findFileByAppName` make a case sensitive search of the name. So when the name in the package.json is different than the name of your folder, the search fail and use `appDelegatePaths[0]` by default

There is three possible steps to resolve this:

- Make the search case sensitive (but maybe you don't want that)
- Don't use `appDelegatePaths[0]` by default when the search fail. In my case the right file was appDelegatePaths[appDelegatePaths.length - 1] (or at least output a warning)
- Make a better error message. The current one is "Could not find line "NSURL \*jsCodeLocation;" in file AppDelegate.m.". It should indicate the full path of the AppDelegate.m that was used to avoid confusion

**Comment by dhei** [comment]
@Sylvain-L Thanks for creating this issue and the investigation. I am curious why your project has difference case between the `name` property of `package.json` and `AppDelegate.m` file path. npm recommend lower case and dash-separated naming convention for `name` property. On the other hand, `AppDelegate.m` file path is created by `react-native init {appname}` which follows the case of appname.

For the three possible mitigation you proposed, I actually think it make sense to search case insensitively. And we will also improve the error message.

Thanks.

**Comment by Sylvain-L** [comment]
@dhei Sorry, I can't satisfy your curiosity, the project is more than one year old and I don't remember how it was created. That is why I started my issue with "Maybe it's bad practice on our part". It feels that we made a little mistake at some point :)

**Comment by bmourat** [comment]
Hi @Sylvain-L 

I'm curious why your appDelegatePaths contains multiple values, do you have more than one AppDelegate.m file? And could you please provide the content of this array for your project?

Thanks,
Murat

**Comment by Sylvain-L** [comment]
Hi @bmourat 

There is multiple match from `ios/Frameworks/Samples/FBAudienceNetwork/*`

These were created by adding signin via facebook. 

@dhei Git to the rescue. The project was indeed initiated with react-native init, using an UpperCamelCase name. The name ion package.json was then manually changed to fit the expected pattern.

**Comment by dhei** [comment]
@Sylvain-L thanks for the information, the case sensitivity fix (#213) will be released in next SDK release.

**Comment by dhei** [comment]
FYI, this is fixed v1.2.0 release, so closing this issue.
